docs/usage/annotations.rst

```diff
@@ -144,3 +145,77 @@ To do so, one can declare an alias to the reusable qubits.
 .. note::
     Even if the reusable alias is not used in code, an alias must be defined to mark qubits as reusable.
     The identifier is insignificant.
+
+Dirty Ancilla Qubits
+--------------------
+
+If qubits are used in a program and are neither marked as output nor reusable, they are considered dirty ancilla qubits thereafter.
+These qubits may be in an arbitrary state, including entanglement with other qubits, and require the explicit `@leqo.dirty` annotation to indicate their intended use in another snippet.
+A dirty ancilla may be promoted to a reusable ancilla via an associated uncomputation block.
+
+To use dirty ancillae within a snippet, the programmer must explicitly opt in by annotating the qubit definition with `@leqo.dirty`.
+
+* The `@leqo.dirty` annotation follows the same implementation rules as input definitions, but omits indexing, as defined in :ref:`input definition <input-anker>`
+
+.. warning::
+    The state of a dirty ancilla qubit can be altered temporarily but must be restored at the end of a snippet.
+    Therefore measuring a dirty qubit is not permitted.
+
+.. code-block:: openqasm3
+    :linenos:
+
+    // Single dirty ancilla
+    @leqo.dirty
+    qubit singleDirtyAncilla;
+
+.. code-block:: openqasm3
+    :linenos:
+
+    // Dirty ancilla array
+    @leqo.dirty
+    qubit[<<length>>] dirtyAncillaArray;
+
+Uncomputation
+~~~~~~~~~~~~~
```

Comment:
Did you set the level of this heading intentionally?
![image](https://github.com/user-attachments/assets/7d68230d-306f-400e-bd63-c0f5336584d3)